### PR TITLE
Make OpenMM an optional dependency

### PR DESCRIPTION
**Pull request by mattwthompson**
## Description
The `simtk` namespace is now more than two years deprecated; it's long past a point in which it makes sense to worry about backwards compatibility for future releases.

Grepping through just instances of `simtk` also surfaced some cases in which things were probably using the wrong registry. I'll see how the tests behaves.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] Remove `simtk` import guards

## Questions
- [ ] Question1

## Status
- [ ] Ready to go

### CODE REVIEW

File: openff/qcsubmit/datasets/datasets.py [hunk 1]
```diff
@@ -497,10 +497,7 @@ def _get_missing_basis_coverage(
 
         import basis_set_exchange as bse
 
-        try:
-            from openmm.app import Element
-        except ImportError:
-            from simtk.openmm.app import Element
+        from openff.qcsubmit.validators import SYMBOLS_TO_ELEMENTS
```
Comment (dotsdl) [inline openff/qcsubmit/datasets/datasets.py:500]: :raised_hands:

**Comment by mattwthompson** [comment]
Bit of a mistake at my part - I spent a chunk of time wondering why I couldn't reproduce the earlier CI failures locally ... only to notice I just hadn't pushed my recent changes. 🤦‍♂️

**Comment by mattwthompson** [comment]
This is ready to go % #245

**Review by j-wags — APPROVED** [review]
Awesome, thanks @mattwthompson! I'll update the releasenotes and merge shortly.

**Pull request merged**